test/common/tmpdir.js

```diff
@@ -100,8 +15,14 @@ const tmpdirName = '.tmp.' +
 const tmpPath = path.join(testRoot, tmpdirName);
 
 let firstRefresh = true;
-function refresh(opts = {}) {
-  rimrafSync(this.path, opts);
+function refresh() {
+  try {
+    fs.rmdirSync(this.path, { recursive: true });
```

Comment:
@trott could setting `maxBusyTries` here help the issue with Windows in the CI?

Reply:
I thought I tried that and it seemed like maybe it was ignored in the synchronous version?

Reply:
Yep. It looks like all of the retry logic is in the async version.

Reply:
Is the solution to pull the retry logic into the sync version as well? @isaacs was there a reason it only exists in the `async` version?

Reply:
So it turns out there is some hard coded retry logic for the sync version: https://github.com/nodejs/node/blob/f17111280e34756c4c8a2d5e7825379baff78e78/lib/internal/fs/rimraf.js#L214-L218. I'm guessing that part of the issue with synchronous retry logic is that there is no synchronous `setTimeout()` (Node would need a sleep API I think).

Reply:
Could we move some of the sync implementation into C and use `usleep`, or just expose sleep for internal use only?

I'm guessing it's an intentional design decision that Node has no `sleep`?

Reply:
Two ideas, but I think they're both problematic:

* We do have `common.busyLoop()` so if we put some retry logic in `common/tmpdir.js`, that could be used. Seems like a suboptimal hack, though.

* Since this is all happening on cleanup, we could also decide to remove the cleanup functionality. The upside is that if someone wants to check something about the tmp files in an `exit` handler, they don't have to worry about another `exit` handler removing the tmp files before they get a chance to do the check. The downside, of course, is that we'll have all these tmp directories left lying around.

I don't think either of those are the way to go, but if they help anyone else arrive at a better idea, cool.

Reply:
> I'm guessing it's an intentional design decision that Node has no sleep?

I don't know if it's intentional or not. libuv has a `uv_sleep()` function in its test runner. I can look into exposing it. I think `sleep()` is a reasonable thing for Node to have outside of this use case.

Reply:
@cjihrig I think this could be great :+1: and we start to gain some benefits from pulling `rimraf` in to core :smile: